docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,320 @@
+
+### Factory data format
+
+The factory data set should be saved into a HEX file that can be written to the flash memory of the Matter device.
+
+In the nRF Connect example, the factory data set is represented in the CBOR format that is stored in a HEX file and then programmed into a device. Two separate scripts are used to create the factory data and the JSON format is used as an intermediate, human-readable representation of the data regulated by the [JSON Schema](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/nrfconnect/nrfconnect_factory_data.schema) file.
+
+All parameters of the factory data set are either mandatory or optional:
+
+- Mandatory parameters must always be provided, as they are required for example to perform commissioning to the Matter network.
+- Optional parameters can be used for development and testing purposes. For example, the ``user`` data parameter consists of all data that is needed by a specific manufacturer and that is not included in the mandatory parameters.
+
+In the factory data set, the following formats are used:
+
+- uint16 and uint32 -- These are the numeric formats representing, respectively, two-bytes length unsigned integer and four-bytes length unsigned integer. This value is stored in a HEX file in the big-endian order.
+- Byte string - This parameter represents the sequence of integers between ``0`` and ``255``(inclusive), without any encoding. Because the JSON format does not allow to use of byte strings, the ``hex:`` prefix is added to a parameter, and its representation is converted to a HEX string. For example, an ASCII string *abba* is represented as *hex:61626261* in the JSON file and then stored in the HEX file as *0x61626261*. The HEX string length in the JSON file is two times greater than the byte string plus the size of the prefix.
+- ASCII string is a string representation in ASCII encoding without null-terminating.
+- ISO 8601 format is a [date format](https://www.iso.org/iso-8601-date-and-time-format.html) that represents a date given as ``YYYY-MM-DD`` or ``YYYYMMDD``.
+- All certificates stored in factory data are given in the [X.509](https://www.itu.int/rec/T-REC-X.509-201910-I/en) format.
+
+<hr>
+<a name="Using own factory data"></a>
+
+## Using own factory data implementation
+
+
+
+<hr>
+<a name="Generating factory data"></a>
+
+## Generating factory data
+
+### Creating factory data JSON file
+
+### Verifying using the JSON Schema tool
+
+The JSON file that contains factory data can be verified using the [JSON Schema tool](https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/nrfconnect/nrfconnect_factory_data.schema). This tool validates the structure and contents of the JSON data. Use this tool to validate the factory data JSON file.
```

Comment:
`JSON Schema tool` links to the shema file not the tool itself. Is that correct?

Reply:
Maybe we can mention that actually there are a few online tools which do the job as well, for instance:
https://www.jsonschemavalidator.net/

Reply:
I think using the online tool here is not a good point, but you're right, there should be only JSON Schema because a tool is described below. I prefer using the python tool instead of online validators, but do you think it is safe and proper to add information that users can use online validators?

Reply:
Well the advantage is that it does not require any pre-setup whatsoever and is easy to use. But regarding the safety and reliability of these tools I have no strong opinion. I just used them in the past quite often and did not faced any issues.